api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java

```diff
@@ -0,0 +1,74 @@
+package org.pmiops.workbench.interceptors;
+
+import com.google.api.client.http.HttpResponseException;
+import com.google.api.services.oauth2.model.Userinfoplus;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.pmiops.workbench.auth.UserInfoService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Service;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+/**
+ * Intercepts all non-OPTIONS API requests to ensure they have an appropriate auth token.
+ */
+@Service
+public class AuthInterceptor extends HandlerInterceptorAdapter {
+
+  private static final String PMI_OPS_SUFFIX = "@pmi-ops.org";
+
+  private static final Logger log = Logger.getLogger(AuthInterceptor.class.getName());
+
+  private final UserInfoService userInfoService;
+
+  @Autowired
+  public AuthInterceptor(UserInfoService userInfoService) {
+    this.userInfoService = userInfoService;
+  }
+
+  @Override
+  public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
+      throws Exception {
+    if (request.getMethod() == HttpMethod.OPTIONS.name()) {
```

Comment:
No, that won't work, will it. :)

I could have sworn that previously sketchycat was making an OPTIONS request which failed here and this logic avoided it. But now I can't reproduce that. Anyway, removing this logic.